source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -71,8 +71,10 @@ void generateV2Header(const std::string& src_addr, const std::string& dst_addr,
         Network::Address::Ipv4Instance(src_addr, src_port).ip()->ipv4()->address();
     const auto net_dst_addr =
         Network::Address::Ipv4Instance(dst_addr, dst_port).ip()->ipv4()->address();
-    memcpy(addrs, &net_src_addr, 4);     // NOLINT(safe-memcpy)
-    memcpy(&addrs[4], &net_dst_addr, 4); // NOLINT(safe-memcpy)
+    safeMemcpy(reinterpret_cast<uint8_t(*)[4]>(addrs),
+               reinterpret_cast<const uint32_t*>(&net_src_addr));
```

Comment:
You don't need the cast from `net_src_addr` or `net_dst_addr`, and I would also change the `const auto`s above to `const uint32_t` to make this more obvious.  See https://github.com/envoyproxy/envoy/blob/7985d815e49f7bd899998436f615ae5497bfff8c/include/envoy/network/address.h#L35 .